File: .agents/skills/open-knowledge-format/scripts/generate_package_digests.rb
#!/usr/bin/env ruby

require 'pathname'
require_relative 'package_digest'

workspace_root = Pathname(ARGV.fetch(0, '.')).expand_path
abort "workspace root does not exist: #{workspace_root}" unless workspace_root.directory?

OkfPackageDigest.workspace_packages(workspace_root).each do |name, package_root|
  puts [name, OkfPackageDigest.digest(package_root), package_root.relative_path_from(workspace_root)].join("\t")
end

File: .agents/skills/open-knowledge-format/scripts/generate-package-digests.mjs
#!/usr/bin/env node

import path from 'node:path';
import { pathToFileURL } from 'node:url';

import { packageDigest, workspacePackages } from './package-digest.mjs';

export async function generatePackageDigests(workspaceRoot = '.') {
  const absoluteRoot = path.resolve(workspaceRoot);
  const packages = await workspacePackages(absoluteRoot);
  return Promise.all(
    packages.map(async ([name, packageRoot]) => [
      name,
      await packageDigest(packageRoot),
      path.relative(absoluteRoot, packageRoot).split(path.sep).join('/'),
    ]),
  );
}

if (isMainModule(import.meta.url)) {
  const workspaceRoot = path.resolve(process.argv[2] ?? '.');
  try {
    for (const fields of await generatePackageDigests(workspaceRoot)) process.stdout.write(`${fields.join('\t')}\n`);
  } catch (error) {
    process.stderr.write(`${error instanceof Error ? error.message : String(error)}\n`);
    process.exitCode = 1;
  }
}

function isMainModule(url) {
  return process.argv[1] !== undefined && url === pathToFileURL(path.resolve(process.argv[1])).href;
}
